assets/src/edit-story/components/library/panes/text/textSets/test/textSets.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import LibraryContext from '../../../../context';
+import TransformContext from '../../../../../transform/context';
+import FontContext from '../../../../../../app/font/context';
+import APIContext from '../../../../../../app/api/context';
+import ConfigContext from '../../../../../../app/config/context';
+import { renderWithTheme } from '../../../../../../testUtils';
+import StoryContext from '../../../../../../app/story/context';
+import TextSets from '../textSets';
+
+function setup() {
+  const libraryValue = {
+    actions: {
+      insertElement: jest.fn(),
+    },
+  };
+  const transformValue = {
+    actions: {
+      registerTransformHandler: () => {},
+    },
+  };
+  const configValue = { api: { stories: [] } };
+  const fontsValue = {
+    actions: {
+      maybeEnqueueFontStyle: () => {},
+    },
+  };
+  const apiValue = {
+    actions: {
+      getAllFonts: () => [],
+    },
+  };
+
+  const storyValue = {
+    actions: {
+      setSelectedElementsById: jest.fn(),
+    },
+  };
+
+  const { queryByText, queryAllByRole } = renderWithTheme(
+    <TransformContext.Provider value={transformValue}>
+      <ConfigContext.Provider value={configValue}>
+        <APIContext.Provider value={apiValue}>
+          <StoryContext.Provider value={storyValue}>
+            <FontContext.Provider value={fontsValue}>
+              <LibraryContext.Provider value={libraryValue}>
+                <TextSets />
+              </LibraryContext.Provider>
+            </FontContext.Provider>
+          </StoryContext.Provider>
+        </APIContext.Provider>
+      </ConfigContext.Provider>
+    </TransformContext.Provider>
```

Comment:
🤯

Reply:
(maybe we should do something about this at some point? This is getting out of hand... @swissspidy?)

Reply:
One idea just occured to me: We could create a central `<DummyProvider provide="tcasfl" />` with the provide prop being a string of characters of the required provided contexts - here it's the exact same sequence as above. And they happen to be unique letters actually, that might not always be true.

We might get a clash between `layout` and `library` and also `snackbar` and `story`, but otherwise, this might work - and we can solve those with capital letters for one of them (or a different letter).

We could of course be explicit with `<DummyProvider provide={["transform","config","api","story","font","library"]} />` but I wanted it to be extremely compact for quicker typing and a better overview.

Thoughts, @swissspidy, @miina, @carlos-kelly, @BrittanyIRL, @mariano-formidable?